extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -161,6 +167,30 @@ export class AmpShareTracking extends AMP.BaseElement {
     }
     return bytes;
   }
+
+  /**
+   * Generate the new url fragment by replacing incoming share tracking fragment
+   * with outgoing share tracking fragment.
+   * The original fragment is not modified.
+   * @param {string} originalViewerFragment
+   * @param {string} incomingFragment
+   * @param {string} outgoingFragment
+   * @return {string}
+   * @private
+   */
+  getNewViewerFragment_(originalViewerFragment, incomingFragment,
+      outgoingFragment) {
+    if (originalViewerFragment == '') {
+      return '.' + outgoingFragment;
+    }
+    if (incomingFragment == '') {
+      return '.' + outgoingFragment + '&' + originalViewerFragment;
+    } else {
+      return originalViewerFragment.replace('.' + incomingFragment, '.' +
+          outgoingFragment);
+    }
+  }
```

Comment:
That'll have an off-by-one due to the `.` not being a part of `incomingFragment`.